repair/table_check.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "replica/database.hh"
+#include "repair/table_check.hh"
+#include "service/migration_manager.hh"
+
+namespace repair {
+
+future<table_dropped> table_sync_and_check(replica::database& db, service::migration_manager& mm, const table_id& uuid) {
+    if (mm.use_raft()) {
```

Comment:
Can we do something for non raft schema as well? E.g., trigger a schema request from the peer node. So this works for both raft and non raft mode?

This is not a must. It is of low priority.  This PR is good enough.

Reply:
I think raft-schema (consistent-cluster-management) is non-optional at this point, so we shouldn't bother working with it.
@kbr-scylla can people disable consistent-cluster-management post 6.0?

Reply:
They can enter RECOVERY mode (https://opensource.docs.scylladb.com/stable/architecture/raft.html#raft-manual-recovery-procedure) (useful when you lose raft majority for example) where cluster operates in gossip mode

Reply:
Do we allow repair+dropping table during recovery mode? :)

Reply:
I know users will not care much about what we allow and go ahead and do stuff anyway. But maybe we don't need to be graceful in this case, we just need to be correct, and failing a repair is fine IMO.

Reply:
Agreed. If this code is called only in this specific scenario (table drop during repair), then it probably can assume Raft. (but a sanity check is still worth it -- if not use raft, throw)

Reply:
Currently it falls-back to just checking locally. I think this is good enough.